client/dashboard/app/hooks/site-logs-data.ts

```diff
@@ -0,0 +1,76 @@
+import { useQuery } from '@tanstack/react-query';
+import { LogType, PHPLog, ServerLog, FilterType } from '../../data/site-logs';
+import { siteLogsQuery } from '../queries/site-logs';
+import type { View } from '@wordpress/dataviews';
+
+const EMPTY_ARRAY: ( ServerLog | PHPLog )[] = [];
+
+const hasLogTypeChanged = ( logType: LogType, data: { logs?: ( PHPLog | ServerLog )[] } ) => {
+	if (
+		logType === 'php' &&
+		!! data.logs &&
+		data.logs.length > 0 &&
+		'severity' in data.logs[ 0 ] &&
+		'message' in data.logs[ 0 ]
+	) {
+		return false;
+	}
+	if (
+		logType === 'server' &&
+		!! data.logs &&
+		data.logs.length > 0 &&
+		'request_type' in data.logs[ 0 ] &&
+		'status' in data.logs[ 0 ]
+	) {
+		return false;
+	}
+	return true;
+};
+
+export function useSiteLogsData( {
+	siteId,
+	view,
+	logType,
+	startTime,
+	endTime,
+	filter,
+}: {
+	siteId: number;
+	view: View;
+	logType: LogType;
+	startTime: number;
+	endTime: number;
+	filter: FilterType;
+} ) {
+	const params: SiteLogsParams = {
+		logType,
+		start: startTime,
+		end: endTime,
+		filter,
+		sortOrder: view.sort?.direction,
+		pageSize: view.perPage,
+		pageIndex: view.page,
+	};
+
+	const { data: siteLogs, isLoading, isFetching } = useQuery( siteLogsQuery( siteId, params ) );
```

Comment:
If the default page size is 50. The logging endpoint feels a lot faster than the "all sites" endpoint. I wonder if it is fast enough that we should preload the first page of site logs in the router. That way we wouldn't need to deal with a `isLoading` state and the user wouldn't be subjected to a spinner 🤔

Reply:
Breaking up the PR sounds great to me